sources/api/apiserver/src/server/mod.rs

```diff
@@ -354,6 +373,153 @@ async fn get_configuration_files(
+            None => status.signal().unwrap_or(1),
+        };
+        Err(match_update_dispatcher_errors(exit_status, None))
+    }
+}
+
+/// Prepares update by downloading the images to the staging partition set
+async fn prepare_update() -> Result<HttpResponse> {
+    let status = Command::new("/usr/bin/thar-be-updates")
+        .arg("prepare-update")
+        .status()
+        .context(error::UpdateDispatcher)?;
+    if status.success() {
+        Ok(HttpResponse::NoContent().finish())
+    } else {
+        let exit_status = match status.code() {
+            Some(code) => code,
+            None => status.signal().unwrap_or(1),
+        };
+        Err(match_update_dispatcher_errors(exit_status, None))
+    }
+}
+
+/// "Activates" an already staged update by bumping the priority bits on the staging partition set
+async fn activate_update() -> Result<HttpResponse> {
+    let output = Command::new("/usr/bin/thar-be-updates")
+        .arg("activate-update")
+        .output()
+        .context(error::UpdateDispatcher)?;
+    if output.status.success() {
+        Ok(HttpResponse::NoContent().finish())
+    } else {
+        let exit_status = match output.status.code() {
+            Some(code) => code,
+            None => output.status.signal().unwrap_or(1),
+        };
+        Err(match_update_dispatcher_errors(
+            exit_status,
+            Some(output.stderr),
+        ))
+    }
+}
+
+/// "Deactivates" an already activated update by rolling back actions done by 'activate-update'
+async fn deactivate_update() -> Result<HttpResponse> {
+    let output = Command::new("/usr/bin/thar-be-updates")
+        .arg("deactivate-update")
+        .output()
+        .context(error::UpdateDispatcher)?;
+    if output.status.success() {
+        Ok(HttpResponse::NoContent().finish())
+    } else {
+        let exit_status = match output.status.code() {
+            Some(code) => code,
+            None => output.status.signal().unwrap_or(1),
+        };
+        Err(match_update_dispatcher_errors(
+            exit_status,
+            Some(output.stderr),
+        ))
+    }
+}
+
+/// Gets the list of available updates and their detailed information
+async fn get_available_updates() -> Result<AvailableUpdatesResponse> {
+    let output = Command::new("/usr/bin/thar-be-updates")
+        .arg("get-available-updates")
+        .output()
+        .context(error::UpdateDispatcher)?;
+    if output.status.success() {
+        let resp: Vec<update_metadata::Update> =
+            serde_json::from_slice(&output.stdout).context(error::UpdateInfoParse {
+                stdout: output.stdout,
+            })?;
+        Ok(AvailableUpdatesResponse(resp))
+    } else {
+        let exit_status = match output.status.code() {
+            Some(code) => code,
+            None => output.status.signal().unwrap_or(1),
+        };
+        Err(match_update_dispatcher_errors(
+            exit_status,
+            Some(output.stderr),
+        ))
+    }
+}
+
```

Comment:
Another way to do this might be to have `thar-be-updates` put all the data in `/run/updates.status` and we can pull out the parts we need for the API response.

Reply:
The portion of the code and the functionality it supports has since then been removed so marking this as resolved.